include/sisl/grpc/rpc_client.hpp

```diff
@@ -338,6 +418,7 @@ class GrpcAsyncClient : public GrpcBaseClient {
             data->m_resp_reader_ptr = (m_stub.get()->*method)(&data->context(), request, cq());
             // CQ tag posted here
             data->m_resp_reader_ptr->Finish(&data->reply(), &data->status(), (void*)data);
```

Comment:
Can we extract the equivalent logic from this and the cb version? even if it's just a localized macro?